logging/fields.go

```diff
@@ -0,0 +1,167 @@
+	"github.com/bloxapp/ssv/protocol/v2/blockchain/beacon"
+	protocolp2p "github.com/bloxapp/ssv/protocol/v2/p2p"
+)
+
+const (
+	FieldAddress             = "address"
+	FieldBindIP              = "bindIP"
+	FieldBlockCacheMetrics   = "blockCacheMetricsField"
+	FieldCurrentSlot         = "currentSlot"
+	FieldDurationNano        = "durationNano"
+	FieldEnr                 = "enr"
+	FieldEventID             = "eventID"
+	FieldFromBlock           = "fromBlock"
+	FieldIdentifier          = "identifier"
+	FieldIndexCacheMetrics   = "indexCacheMetrics"
+	FieldMessageID           = "messageID"
+	FieldPeerID              = "peerID"
+	FieldPrivateKey          = "privKey"
+	FieldPubKey              = "pubKey"
+	FieldResults             = "results"
+	FieldStartTimeUnixNano   = "startTimeUnixNano"
+	FieldSubnets             = "subnets"
+	FieldSyncOffset          = "syncOffset"
+	FieldTargetNodeEnr       = "targetNodeEnr"
+	FieldTxHash              = "txHash"
+	FieldUpdatedEnrLocalNode = "updated_enr"
+	FieldValidator           = "validator"
+)
+
+func FromBlock(val fmt.Stringer) zapcore.Field {
+	return zap.Stringer(FieldFromBlock, val)
+}
+
+func SyncOffset(val fmt.Stringer) zapcore.Field {
+	return zap.Stringer(FieldSyncOffset, val)
+}
+
+func TxHash(val fmt.Stringer) zapcore.Field {
+	return zap.Stringer(FieldTxHash, val)
+}
+
+func EventID(val fmt.Stringer) zapcore.Field {
+	return zap.Stringer(FieldEventID, val)
+}
+
+func Identifier(val fmt.Stringer) zapcore.Field {
+	return zap.Stringer(FieldIdentifier, val)
+}
+
+func IdentifierBytes(val []byte) zapcore.Field {
+	return zap.Stringer(FieldIdentifier, hexStringer{val})
+}
+
+func PubKey(val []byte) zapcore.Field {
+	return zap.Stringer(FieldPubKey, hexStringer{val})
+}
+
+func PrivKey(val []byte) zapcore.Field {
+	return zap.Stringer(FieldPrivateKey, hexStringer{val})
+}
+
+func Validator(val []byte) zapcore.Field {
+	return zap.Stringer(FieldValidator, hexStringer{val})
+}
+
+func Address(val url.URL) zapcore.Field {
+	return zap.Stringer(FieldAddress, &val)
+}
+
+func Enr(val *enode.Node) zapcore.Field {
+	return zap.Stringer(FieldEnr, val)
+}
+
+func TargetNodeEnr(val *enode.Node) zapcore.Field {
+	return zap.Stringer(FieldTargetNodeEnr, val)
+}
+
+func EnrLocalNode(val *enode.LocalNode) zapcore.Field {
+	return zap.Stringer(FieldEnr, val.Node())
+}
+
+func UpdatedEnrLocalNode(val *enode.LocalNode) zapcore.Field {
+	return zap.Stringer(FieldUpdatedEnrLocalNode, val.Node())
+}
+
+func Subnets(val records.Subnets) zapcore.Field {
+	return zap.Stringer(FieldSubnets, val)
+}
+
+func PeerID(val peer.ID) zapcore.Field {
+	return zap.Stringer(FieldPeerID, val)
+}
+
+func BindIP(val net.IP) zapcore.Field {
+	return zap.Stringer(FieldBindIP, val)
+}
+
+func MessageID(val spectypes.MessageID) zapcore.Field {
+	return zap.Stringer(FieldMessageID, val)
+}
+
+func DurationNano(val time.Time) zapcore.Field {
+	return zap.Stringer(FieldDurationNano, int64Stringer{time.Since(val).Nanoseconds()})
```

Comment:
why nano?

Reply:
it's the smallest available value. for example if you used milliseconds and are debugging a timing issue in prod, it can appear that something is happening at the same time because the actions happened in the same millisecond, but in reality one action preceded the other by a duration smaller than one millisecond